src/mscorlib/shared/System/Convert.cs

```diff
@@ -2658,83 +2658,132 @@ public static bool TryFromBase64String(string s, Span<byte> bytes, out int bytes
 
-                bytesWritten += moreBytesWritten;
-                return true;
-            }
-            else
-            {
-                // If we got here, the last valid 4-character chunk read had padding characters ('='). Hence, we cannot accept 
-                // any more Base64 characters after this point but we can accept trailing whitespace.
-                while (consumed < chars.Length)
+                chars = chars.Slice(consumedInThisIteration);
+                bytes = bytes.Slice(bytesWrittenInThisIteration);
+
+                Debug.Assert(chars.Length != 0); // If TryDecodeFromUtf16() consumed the entire buffer, it could not have returned false.
+                if (chars[0].IsSpace())
                 {
-                    char c = chars[consumed++];
-                    if (c != ' ' && c != '\n' && c != '\r' && c != '\t')
+                    // If we got here, the very first character not consumed was a whitespace. We can skip past any consecutive whitespace, then continue decoding.
+
+                    int indexOfFirstNonSpace = 1;
+                    for (; ; )
+                    {
+                        if (indexOfFirstNonSpace == chars.Length)
+                            break;
+                        if (!chars[indexOfFirstNonSpace].IsSpace())
+                            break;
+                        indexOfFirstNonSpace++;
+                    }
+
+                    chars = chars.Slice(indexOfFirstNonSpace);
+
+                    if ((bytesWrittenInThisIteration % 3) != 0 && chars.Length != 0)
                     {
+                        // If we got here, the last successfully decoded block encountered an end-marker, yet we have trailing non-whitespace characters.
+                        // That is not allowed.
                         bytesWritten = default;
                         return false;
                     }
+
+                    // We now loop again to decode the next run of non-space characters. 
+                }
+                else
+                {
+                    unsafe
+                    {
+                        Debug.Assert(chars.Length != 0 && !chars[0].IsSpace());
+
+                        // If we got here, it is possible that there is whitespace that occurred in the middle of a 4-byte chunk. That is, we still have
+                        // up to three Base64 characters that were left undecoded by the fast-path helper because they didn't form a complete 4-byte chunk.
+                        // This is hopefully the rare case (multiline-formatted base64 message with a non-space character width that's not a multiple of 4.)
+                        // We'll filter out whitespace and copy the remaining characters into a temporary buffer.
+                        TempCharBuffer tempBufferStorage = default;
+                        Span<char> tempBuffer = MemoryMarshal.CreateSpan(ref tempBufferStorage.C0, sizeof(TempCharBuffer) / sizeof(char));
```

Comment:
Can we use a stackalloc span here instead of a struct of chars?

Reply:
Wouldn't that cause a new (stack) allocation each time through the loop?

Reply:
Good point. Given this is deeply nested, it is probably not worth adding a stackalloc span right at the top and re-use that (since we would incur the cost of creating the stackalloc span when not necessary).

Reply:
The JIT will convert small fixed stackallocs outside the loops into regular local. I think you can just do a safe stackalloc at the top of them method without a problem. It should generate same or even better code than what you have now.